examples/jetstream/workqueue-stream/java/Main.java

```diff
@@ -0,0 +1,171 @@
+
+      // Define the stream configuration, specifying the retention policy
+      // as `WorkQueue` and create it.
+      StreamConfiguration sc = StreamConfiguration.builder()
+        .name("EVENTS")
+        .subjects("events.>")
+        .retentionPolicy(RetentionPolicy.WorkQueue)
+        .build();
+
+      jsm.addStream(sc);
+      System.out.printf("Created stream %s\n", sc.getName());
+
+      // ### Queue
+      // Create a `JetStream` context for publishing and subscribing.
+      JetStream js = nc.jetStream();
+
+      js.publish("events.us.page_loaded", null);
+      js.publish("events.eu.mouse_clicked", null);
+      js.publish("events.us.input_focused", null);
+      System.out.println("Published 3 messages");
+
+      StreamInfo info = jsm.getStreamInfo("EVENTS");
+      long msgCount = info.getStreamState().getMsgCount();
+      System.out.printf("msgs in stream: %s\n", msgCount);
+
+      // ### Adding a consumer
+      // Now let's add a consumer. It can be a *push* or *pull* consumer,
+      // however for this example we will use *pull*.
+      // The first pull consumer will be bound to all subjects in
+      // the stream. 
+      ConsumerConfiguration c1 = ConsumerConfiguration.builder()
+        .durable("processor-1")
+        .ackPolicy(AckPolicy.Explicit)
+        .build();
+
+      jsm.addOrUpdateConsumer("EVENTS", c1);
+
+      // Now we bind the subscription. Note the empty string for the subject
+      // implies interest in all subjects in the stream.
+      PullSubscribeOptions o1 = PullSubscribeOptions.bind("EVENTS", "processor-1");
+      JetStreamSubscription s1 = js.subscribe("", o1);
+
+      // Now we can fetch all three messages in the stream.
+      s1.pull(3);
+      for (int i = 0; i < 3; i++) {
+        Message msg = s1.nextMessage(Duration.ofSeconds(1));
+        msg.ack();
+        System.out.printf("Received event: %s\n", msg.getSubject());
+      }
+      s1.unsubscribe();
+
+      // We can confirm by checking the stream state again.
+      info = jsm.getStreamInfo("EVENTS");
+      msgCount = info.getStreamState().getMsgCount();
+      System.out.printf("msgs in stream: %s\n", msgCount);
+
+      // ### Exclusive non-filtered consumer
+      // As noted in the description above, work-queue streams can only
+      // have at most one consumer with interest on a subject at any given
+      // time. Since this first pull consumer is not filtered, it we try
+      // to create another consumer, it will fail.
+      ConsumerConfiguration c2 = ConsumerConfiguration.builder()
+        .durable("processor-2")
+        .ackPolicy(AckPolicy.Explicit)
+        .build();
+
+      try {
+        jsm.addOrUpdateConsumer("EVENTS", c2);
+      } catch (JetStreamApiException e) {
+        System.out.println("Failed to create an overlapping consumer");
+      }
+
+      // However if we delete the first consumer, this will clean up
+      // the first consumer since it was an ephemeral, which removes interest.
```

Comment:
not an ephemeral, I think you can remove `since it was an ephemeral`

Reply:
Fixed, wrote that before updating to use durables.